controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -126,6 +127,18 @@ func (r *ReplicatedPolicyReconciler) Reconcile(ctx context.Context, request ctrl
 		return reconcile.Result{}, nil
 	}
 
+	// Handle replicated policy, which is related to policyset
+	isHandled, err := r.handlePolicySet(ctx, rootPolicy, replicatedPolicy)
```

Comment:
Is this actually needed? I think the code below should notice the case where the policy only had a placement bound through the policy set and clean it up appropriately.

Reply:
This func cleanup replicated policy connected to policyset... without this, breaks e2e test. I moved all placement watch from root to replicated policy. that is why happen

Reply:
Which e2e test breaks? I basically disabled this bit, and ran the tests and they seemed to pass (one is re-running for a PolicyAutomation test that mysteriously failed): https://github.com/JustinKuli/governance-policy-propagator-1/pull/10

Reply:
What I would expect is that when a PolicySet is deleted, a reconcile is queued for each of the root policies in that set. Each of those then figures out all of the clusters the policy should be on, and when it does that it would notice that the PolicySet was gone, and not consider the placements that were previously associated with that set.